### PR TITLE
US113177: Manual alignment option for points or custom points based rubrics

### PR DESCRIPTION
**Pull request by lvyuan95-d2l**
Works with LMS PR: https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/8275

### CODE REVIEW

File: editor/d2l-rubric-alignment-editor.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			</template>
+		</div>
+	</template>
+</dom-module>`;
+
+document.head.appendChild($_documentContainer.content);
+Polymer({
+	is: 'd2l-rubric-alignment-editor',
+	properties: {
+		href: {
```
Comment (rgoessl) [inline editor/d2l-rubric-alignment-editor.js]: Don't `href` and `token` props come from one of the behaviours?

File: editor/d2l-rubric-alignment-editor.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	_changeAlignment: function(e) {
+		this._debounceSelected = e.target.value;
+		this.debounce('change', function() {
+			var action = this.entity.getActionByName('update-alignment');
+			if (action) {
+				this.toggleBubble('_alignmentInvalid', false, 'alignment-bubble');
+				var fields = [{ 'name': 'alignment', 'value': this._debounceSelected }];
+				this.performSirenAction(action, fields).then(function() {
+					this.fire('d2l-rubric-alignment-saved');
+					this.fire('d2l-loa-overlay-refresh');
```
Comment (rgoessl) [inline editor/d2l-rubric-alignment-editor.js]: Don't events only get thrown up the DOM? How does the loa component receive this?

File: build/lang/zh.js [hunk 3]
```diff
@@ -79,7 +82,7 @@ D2L.PolymerBehaviors.Rubric.LocalizeBehavior.LangZhBehavior = {
 		'loaLevelUpdatedLabelMultiRubric': '成绩级别“{loaLevelName}”现已映射到量规标准 {otherRubricLevelNames} 和 {lastRubricLevelName}',
 		'loaLevelUpdatedLabelSingleRubric': '成绩级别“{loaLevelName}”现已映射到量规标准 {rubricLevelName}',
 		'loaOverlayHeading': '成绩水平',
-		'loaSliderLabel': 'Achievement threshold between \'{loaLevel1}\' and \'{loaLevel2}\'. Use left and right arrow keys to move the threshold.',
+		'loaSliderLabel': 'Achievement threshold between "{loaLevel1}" and "{loaLevel2}". Use left and right arrow keys to move the threshold.',
```
Comment (rgoessl) [inline build/lang/zh.js]: Need lint?

File: lang/fr.json [hunk 4]
```diff
@@ -1,143 +1,147 @@
...
-   "selectPreviousLevel" : "Sélectionner le niveau précédent",
-   "setScoreVisibilityFailed" : "Le paramétrage de la visibilité des notes a échoué.",
-   "shareRubricSaveFailed" : "Échec de la modification de la disponibilité de l’unité organisationnelle",
-   "sharedAlertText" : "Cette rubrique ne peut être modifiée qu’à partir de l’unité org qui l’a partagée",
-   "statistics" : "Statistiques",
-   "total" : "Total",
-   "totalMobile" : "Résultat total",
-   "totalScoreAriaLabel" : "La grille d’évaluation compte pour la note totale de {value} points.",
-   "totalScoreLabel" : "Note finale de la grille d’évaluation"
+    "addCriteriaGroup": "Ajouter un groupe de critères",
```
Comment (rgoessl) [inline lang/fr.json]: different line endings/tabs?

File: editor/d2l-rubric-alignment-editor.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+	_changeAlignment: function(e) {
+		this._debounceSelected = e.target.value;
+		this.debounce('change', function() {
+			var action = this.entity.getActionByName('update-alignment');
+			if (action) {
+				this.toggleBubble('_alignmentInvalid', false, 'alignment-bubble');
+				var fields = [{ 'name': 'alignment', 'value': this._debounceSelected }];
+				this.performSirenAction(action, fields).then(function() {
+					this.fire('d2l-rubric-alignment-saved');
```
Comment (rgoessl) [inline editor/d2l-rubric-alignment-editor.js]: what does this event do? who catches it?

File: editor/d2l-rubric-alignment-editor.js (hunk 1 above)
Comment (khalilleo) [reply editor/d2l-rubric-alignment-editor.js]: Didn't know about that, but I just followed the pattern like https://github.com/Brightspace/d2l-rubric/blob/master/editor/d2l-rubric-editor.js#L304

File: editor/d2l-rubric-alignment-editor.js (hunk 2 above)
Comment (khalilleo) [reply editor/d2l-rubric-alignment-editor.js]: I thought it's like the global event... but it worked though

File: editor/d2l-rubric-alignment-editor.js (hunk 5 above)
Comment (khalilleo) [reply editor/d2l-rubric-alignment-editor.js]: I'll add a test for this

File: editor/d2l-rubric-alignment-editor.js (hunk 1 above)
Comment (lvyuan95-d2l) [reply editor/d2l-rubric-alignment-editor.js]: Didn't know about that, but I just followed the pattern like https://github.com/Brightspace/d2l-rubric/blob/master/editor/d2l-rubric-editor.js#L304

File: editor/d2l-rubric-alignment-editor.js (hunk 2 above)
Comment (lvyuan95-d2l) [reply editor/d2l-rubric-alignment-editor.js]: I thought it's like the global event... but it worked though

File: editor/d2l-rubric-alignment-editor.js (hunk 5 above)
Comment (lvyuan95-d2l) [reply editor/d2l-rubric-alignment-editor.js]: I'll add a test for this

File: lang/fr.json (hunk 4 above)
Comment (lvyuan95-d2l) [reply lang/fr.json]: Something funky with my VS code...

Comment (lvyuan95-d2l) [reply lang/fr.json]: Isn't it supposed to be 4 spaces?

File: build/lang/zh.js (hunk 3 above)
Comment (lvyuan95-d2l) [reply build/lang/zh.js]: `npm run lang:lint -- --fix` doesn't seem to do anything

Comment (rgoessl) [reply build/lang/zh.js]: Hmmm. Try updating polymer-langtools package. I think did made a fix for this

Comment (lvyuan95-d2l) [reply build/lang/zh.js]: Just did lint fix with `"d2l-polymer-langtools": "^1.1.5"`, but it tried to turn
`"Résultats d'apprentissage"` to
`'Résultats d'apprentissage'` 🤔

File: lang/fr.json (hunk 4 above)
Comment (rgoessl) [reply lang/fr.json]: Not sure

**Pull request merged**